app/dao/templates_dao.py

```diff
@@ -185,16 +188,108 @@ def dao_get_number_of_templates_by_service_id_and_name(
     return db.session.scalar(stmt)
 
 
+class TemplateHistoryData:
+    def __init__(
+        self,
+        id,
+        name,
+        template_type,
+        created_at,
+        updated_at,
+        content,
+        service_id,
+        subject,
+        postage,
+        created_by_id,
+        version,
+        archived,
+        process_type,
+        service_letter_contact_id,
+        content_as_html=None,
+        content_as_plain_text=None,
+        hidden=False,
+        onsite_notification=False,
+        reply_to_email=None,
+        provider_id=None,
+        communication_item_id=None,
+        redact_personalisation=False,
+        get_reply_to_text=None,
+    ):
+        # TemplateHistory attributes
+        self.id = id
+        self.name = name
+        self.template_type = template_type
+        self.created_at = created_at
+        self.updated_at = updated_at
+        self.content = content
+        self.service_id = service_id
+        self.subject = subject
+        self.postage = postage
+        self.created_by_id = created_by_id
+        self.version = version
+        self.archived = archived
+        self.process_type = process_type
+        self.service_letter_contact_id = service_letter_contact_id
+
+        # Additional attributes from TemplateBase
+        self.content_as_html = content_as_html
+        self.content_as_plain_text = content_as_plain_text
+        self.hidden = hidden
+        self.onsite_notification = onsite_notification
+        self.reply_to_email = reply_to_email
+        self.provider_id = provider_id
+        self.communication_item_id = communication_item_id
+        self.redact_personalisation = redact_personalisation
+
+        self.get_reply_to_text = get_reply_to_text
+
+
+@cached(cache=template_history_cache)
+def dao_get_template_history_by_id(template_id, version) -> TemplateHistory | None:
+    stmt = select(TemplateHistory).where(TemplateHistory.id == template_id, TemplateHistory.version == version)
+    template_history_object = db.session.scalars(stmt).one()
```

Comment:
[one](https://docs.sqlalchemy.org/en/14/core/connections.html#sqlalchemy.engine.ScalarResult.one) returns an instance or raises an exception.  If you want an instance or None, use [first](https://docs.sqlalchemy.org/en/14/core/connections.html#sqlalchemy.engine.ScalarResult.first).